lib/sinon/spy.js

```diff
@@ -202,6 +195,13 @@ var spyApi = {
 
         push.call(this.exceptions, exception);
         push.call(this.returnValues, returnValue);
+        if (matchings) {
```

Comment:
As above, assume arrays from `matchingFakes` and prefer ES5.1 array extras over `for` loops.